### PR TITLE
Drain nodes in parallel

### PR DESCRIPTION
**Pull request by aclevername**
### Description

Bring back https://github.com/weaveworks/eksctl/pull/4864

1 new commit added to ensure `delete cluster` sets the parallel limit

To review just look at the latest commit

addon, cluster_dns and existing_vpc integration tests all passing:
```
[0] Ginkgo ran 1 suite in 25m33.42732661s
[0] Test Suite Passed
[0]=================================== Done ===================================
2022/03/17 11:21:05 ================== Summary ================================
[0] Ran 2 of 2 Specs in 1523.222 seconds
[0] SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 0 Skipped
[0] PASS
[0]




[0]
[0] Ginkgo ran 1 suite in 25m35.491334968s
[0] Test Suite Passed
[0]=================================== Done ===================================
2022/03/17 11:21:07 ================== Summary ================================
[0] Ran 1 of 1 Specs in 1525.000 seconds
[0] SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
[0] PASS
[0]


[0]
[0] Ginkgo ran 1 suite in 22m5.420170687s
[0] Test Suite Passed
[0]=================================== Done ===================================
2022/03/17 11:17:37 ================== Summary ================================
[0] Ran 1 of 1 Specs in 1314.836 seconds
[0] SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
[0] PASS
```

### CODE REVIEW

File: pkg/drain/nodegroup.go [hunk 1]
```diff
@@ -127,35 +136,58 @@ func (n *NodeGroupDrainer) Drain() error {
...
+					}
+				}(i, node)
 			}
 		}
 	}
 }
 
+func waitForAllRoutinesToFinish(ctx context.Context, sem *semaphore.Weighted, size int64) {
+	if err := sem.Acquire(ctx, size); err != nil {
+		logger.Critical("failed to claim sem: %w", err)
```
Comment (Skarlso) [inline pkg/drain/nodegroup.go:178]:
```suggestion
		logger.Critical("failed to acquire semaphore while waiting for all routines to finish: %w", err)
```

Just so we know which one of the two failed to acquire a semaphore. :)

File: pkg/drain/nodegroup.go [hunk 2]
```diff
@@ -127,35 +136,58 @@ func (n *NodeGroupDrainer) Drain() error {
...
-				if err != nil {
-					logger.Warning("pod eviction error (%q) on node %s", err, node)
-					time.Sleep(retryDelay)
-					continue
-				}
-				logger.Debug("%d pods to be evicted from %s", pending, node)
-				if pending == 0 {
-					drainedNodes.Insert(node)
+				if err := sem.Acquire(ctx, 1); err != nil {
+					logger.Critical("failed to claim sem: %w", err)
```
Comment (Skarlso) [inline pkg/drain/nodegroup.go:148]:
```suggestion
					logger.Critical("failed to acquire semaphore: %w", err)
```

**Review by Skarlso — APPROVED** [review]
Two nits, then LGTM :) Again :DDDD

**Review by Himangini — APPROVED** [review]
👍🏻

**Pull request merged**